### PR TITLE
Fix/remove third party constant

### PR DESCRIPTION
**Pull request by angrybayblade**
## Proposed changes

Remove the need for `THIRD_PARTY` constant and updates the base image classes to set the third party contracts path on runtime.

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [x] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [ ] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [x] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have locally run services that could be impacted and they do not present failures derived from my changes

### CODE REVIEW

File: autonomy/test_tools/configurations.py [hunk 1]
```diff
@@ -31,8 +31,6 @@ def get_key(key_path: Path) -> str:
 
 CUR_PATH = os.path.dirname(inspect.getfile(inspect.currentframe()))  # type: ignore
 TEST_TOOLS_DIR = Path(CUR_PATH).resolve().absolute()
-ROOT_DIR = Path(CUR_PATH, "..", "..").resolve().absolute()
-THIRD_PARTY = ROOT_DIR / "third_party"
```
Comment (Karrenbelt) [inline autonomy/test_tools/configurations.py:35]: so now we always obtain it the THIRD_PARTY as a constant from the conftest.py file in the respective test directory of whatever repository autonomy is installed in, am I understanding correctly?

this then creates a constraint on any repository using the autonomy package to specify these in their `tests/conftest.py` file?

Comment (angrybayblade) [reply autonomy/test_tools/configurations.py:35]: Not necessarily from the conftest.py, you can set it on the test class like this

https://github.com/valory-xyz/open-autonomy/pull/1199/files#diff-7dbf9570300e3300f3e8c673ea8087d701d596adf755023c02b5035a10c9a803R75

**Review by DavidMinarsch — APPROVED** [review]
Let's also open an issue to fully remove the `contracts-amm` (it should not be part of autonomy) and replace the current gnosis safe third party with our own registry repo, so we have 1:1 parity with our deployments.

File: autonomy/test_tools/configurations.py (hunk 1 above)
Comment (Karrenbelt) [reply autonomy/test_tools/configurations.py:35]: So now we should provide the necessary contracts ourselves in our repository. Would it be an option to link the respective contracts as submodules along with the package?

for example, `GnosisSafeNetDockerImage` replies on `safe-contracts` specifically.

**Review by Adamantios — APPROVED** [review]

**Pull request merged**